### PR TITLE
#1049 wp create make blocked by autocomplete

### PR DESCRIPTION
**Pull request by tigerhippo**
## Changes

In the CreateWorkPackageForm file:
- Copy pasted code relating to the wbsNum from Mihir's PR

In the CreateWorkPackageFormView file:
- Updated code to account for the wbsNum passed in from the CreateWorkPackageForm file 
- Copy pasted FormControl section from Peyton's PR 
- Obtained the project with given wbsNum and then obtained the work packages for the project
- Made function blockedByToAutocompleteOption which shows how each work package option should be displayed in the autocomplete

## To Do

- Fix error (it doesn't seem to be occurring anymore, not sure why)
- yarn test:backend + frontend
- manually test on FinishLine
- Get rid of any comments
- Abstract common features between multiple autocompletes into an NERMultipleAutocomplete

## Checklist

- [ ] All commits are tagged with the ticket number
- [ ] No linting errors / newline at end of file warnings
- [ ] All code follows repository-configured prettier formatting
- [ ] No merge conflicts
- [ ] All checks passing
- [ ] Screenshots of UI changes (see Screenshots section)
- [ ] Remove any non-applicable sections of this template
- [ ] Assign the PR to yourself
- [ ] No `yarn.lock` changes (unless dependencies have changed)
- [ ] Request reviewers & ping on Slack
- [ ] PR is linked to the ticket (fill in the closes line below)

Closes # (issue #)

### CODE REVIEW

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageForm.tsx [hunk 1]
```diff
@@ -12,15 +12,17 @@ import { routes } from '../../utils/routes';
...
 
 export interface CreateWorkPackageFormInputs {
   name: string;
   startDate: Date;
   duration: number | null;
   crId: number;
   stage: WorkPackageStage | 'None';
-  wbsNum: string;
-  blockedBy: { wbsNum: string }[];
+  wbsNum: string | null;
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageForm.tsx:24]: switch this to just string (my b)

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 2]
```diff
@@ -72,17 +87,19 @@ const CreateWorkPackageFormView: React.FC<CreateWorkPackageFormViewProps> = ({ a
     resolver: yupResolver(schema),
     defaultValues: {
       name: '',
-      wbsNum: query.get('wbs') || '',
       crId: Number(query.get('crId')),
       stage: 'NONE' as WorkPackageStage | 'None',
       startDate,
+      wbsNum: null,
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:93]: you can set the default value to the wbsNum that's being passed in as a prop

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageForm.tsx [hunk 3]
```diff
@@ -29,13 +31,14 @@ const CreateWorkPackageForm: React.FC = () => {
   const history = useHistory();
   const auth = useAuth();
   const toast = useToast();
+  const query = useQuery();
 
   const { isLoading, mutateAsync } = useCreateSingleWorkPackage();
-
+  const [wbsNum, setWbsNum] = useState(query.get('wbsNum') || '');
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageForm.tsx:37]: change this to query.get('wbs') (not your bad)

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx [hunk 4]
```diff
@@ -50,19 +49,34 @@ const schema = yup.object().shape({
...
+}) => {
   const startDate = new Date();
   const today = startDate.getDay();
   if (today !== 1) {
     const daysUntilNextMonday = (7 - today + 1) % 7;
     startDate.setDate(startDate.getDate() + daysUntilNextMonday);
   }
   const query = useQuery();
+
+  const { data: project } = useSingleProject(validateWBS(query.get('wbs') || ''));
```
Comment (RChandler234) [inline src/frontend/src/pages/CreateWorkPackagePage/CreateWorkPackageFormView.tsx:77]: replace query.get('wbs') || '' with wbsNum

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Comment by Peyton-McKee** [comment]
other PR for this

**Pull request closed without merging**